python/src/aiconfig/eval/common.py

```diff
@@ -54,25 +54,25 @@ class EvaluationMetricMetadata(cu.Record, Generic[T_OutputDatum]):
           Error count must fall between 0 and inf.
     """
 
-    @property
-    def id(self) -> str:
-        return cu.hash_id(
-            f"{self.name}{self.description}{self.best_value}{self.worst_value}params={self._serialize_extra_metadata()}".encode("utf-8")
-        )
-
-    def _serialize_extra_metadata(self) -> str:
-        return json.dumps(self.extra_metadata, sort_keys=True)
-
     name: str
     description: str
     best_value: MetricValue | None = None
     worst_value: MetricValue | None = None
     # e.g. {"substring": "hello", "case_sensitive": False}
     extra_metadata: dict[str, Any] = {}
+    id: str = "UNSET"
+
+    @model_validator(mode="before")
+    def set_id(cls, values: dict[str, Any]) -> dict[str, Any]:
+        values["id"] = cu.hash_id(f"{values['name']}{values['description']}{values['best_value']}{values['worst_value']}".encode("utf-8"))
+        return values
+
+    def _serialize_extra_metadata(self) -> str:
+        return json.dumps(self.extra_metadata, sort_keys=True)
 
 
 class SampleMetricValue(cu.Record, Generic[T_OutputDatum]):
-    value: MetricValue
+    value: MetricValue | None
```

Comment:
What's the intention of a None value here?

Reply:
Placeholder for errors